azure-mgmt-resource/README.rst

```diff
@@ -54,25 +54,94 @@ future versions.
 Usage
 =====
 
-Examples
---------
+Authentication
+--------------
 
-We'll work on adding some documentation here, but for now, see the following
-for examples on using the ARM APIs.
+Authentication with Azure Resource Manager is done via tokens.
 
--  `Azure Resource Viewer Web Application Sample <https://github.com/Azure/azure-sdk-for-python/tree/master/examples/AzureResourceViewer>`__
--  `Azure Resource Manager Unit tests <https://github.com/Azure/azure-sdk-for-python/tree/master/azure-mgmt/tests>`__
+First, create a service principal using the instructions found
+`here <https://azure.microsoft.com/en-us/documentation/articles/resource-group-create-service-principal-portal/>`__.
 
-Authentication
---------------
+Then, use the following code to obtain an authentication token.
+
+.. code:: python
+
+    import requests
+
+    def get_token_from_client_credentials(endpoint, client_id, client_secret):
+        payload = {
+            'grant_type': 'client_credentials',
+            'client_id': client_id,
+            'client_secret': client_secret,
+            'resource': 'https://management.core.windows.net/',
+        }
+        response = requests.post(endpoint, data=payload).json()
+        return response['access_token']
+
+    # TODO: Replace endpoint, client id and secret for your application
+    # In Azure portal, in your application configure page:
+    # - Click on View Endpoints, use the OAuth 2.0 Token Endpoint
+    # - The client id is already generated for you
+    # - The client secret is only displayed when the key is created the first time
+    auth_token = get_token_from_client_credentials(
+        endpoint='https://login.microsoftonline.com/00000000-0000-0000-0000-000000000000/oauth2/token',
+        client_id='11111111-1111-1111-1111-111111111111',
+        client_secret='2222222222222222222222222222222222222222222=',
+    )
+
+Create the management client
+----------------------------
+
+The following code uses the authentication token obtained in the previous
+section and create an instance of the management client.
+
+.. code:: python
 
-Authentication with ARM is done via tokens.
+    from azure.mgmt.common import SubscriptionCloudCredentials
+    from azure.mgmt.resource import ResourceManagementClient
 
-You can use the `ADAL <https://pypi.python.org/pypi/adal>`__ library to
-obtain authentication tokens.
+    # TODO: Replace this with your subscription id
+    subscription_id = '33333333-3333-3333-3333-333333333333'
```

Comment:
perhaps to simplify this you could put the constants (they are all the same) in their own codeblock and just use them in these?  Will shorten the examples a hair?